File: leetcode/go/9 Palindrome Number.go
func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	magic := []int8{}
	for x != 0 {
		magic = append(magic, int8(x%10))
		x /= 10
	}
	for i := 0; i < len(magic)/2; i++ {
		if magic[i] != magic[len(magic)-1-i] {
			return false
		}
	}
	return true
}

File: leetcode/ts/9 Palindrome Number.ts
function isPalindrome(x: number): boolean {
    if (x < 0) 
        return false
    let div : number = 1
    while (x / div >= 10) div *= 10
    while (x > 0) {
        let left : number = Math.floor(x / div)
        let right = x % 10
        if (left != right) return false
        x = Math.floor((x % div) / 10)
        div = Math.floor(div / 100)
    }
    return true
};